test/matcher_combinators/core_test.clj

```diff
@@ -307,6 +306,12 @@
 (def pred-set #{(pred-matcher odd?) (pred-matcher pos?)})
 (def pred-seq [(pred-matcher odd?) (pred-matcher pos?)])
 
+(def short-equals-seq (map equals [1 3]))
+
+(fact "embeds for sequences"
+  (core/match (embeds short-equals-seq) [3 4 1]) => (just [:match (just [3 4 1])])
+  (core/match (embeds short-equals-seq) [3 4 1 5]) => (just [:match (just [3 4 1 5])]))
```

Comment:
this case wasn't working before because `1` and `3` were never going to be next to each other in the permutations, so the way subset matching was done, this was causing issues